src/pages/Filing/FilingApp/FilingWarnings/index.tsx

```diff
@@ -1,175 +1,226 @@
+import FilingWarningsAlerts, {
+  InstitutionFetchFailAlert,
+} from './FilingWarningsAlerts';
+
+const isSubmissionAccepted = (submission?: SubmissionResponse): boolean => {
+  return submission?.state === FileSubmissionState.SUBMISSION_ACCEPTED;
+};
 
 function FilingWarnings(): JSX.Element {
-  const { lei, year } = useParams();
   const navigate = useNavigate();
+  const auth = useSblAuth();
+  const { lei, year } = useParams();
+  const [boxChecked, setBoxChecked] = useState(false);
+  const [formSubmitLoading, setFormSubmitLoading] = useState(false);
+  const [formSubmitError, setFormSubmitError] = useState(null);
 
   const {
-    isFetching: isFetchingGetSubmissionLatest,
-    error: errorGetSubmissionLatest,
-    data: actualDataGetSubmissionLatest,
+    data: submission,
+    isLoading: isSubmissionLoading,
+    isError: errorSubmissionFetch,
   } = useGetSubmissionLatest({ lei, filingPeriod: year });
 
   const {
     data: institution,
-    isLoading: isLoadingInstitution,
-    isError: isErrorInstitution,
+    isLoading: isInstitutionLoading,
+    isError: errorInstitutionFetch,
   } = useInstitutionDetails(lei);
 
-  const institutionName = isLoadingInstitution
-    ? 'Loading...'
-    : isErrorInstitution
-      ? ''
-      : institution.name;
-
   const formattedData = useMemo(
-    () => getErrorsWarningsSummary(actualDataGetSubmissionLatest),
-    [actualDataGetSubmissionLatest],
+    () => getErrorsWarningsSummary(submission),
+    [submission],
   );
 
+  if (isSubmissionLoading || isInstitutionLoading) return <LoadingContent />;
+
+  /* 
+    Derived data
+  */
   const { logicWarningsMulti, logicWarningsSingle } = formattedData;
 
-  // Determines Alert and if 'Save and continue' button is disabled
-  const errorState = [logicWarningsMulti, logicWarningsSingle].some(
+  const hasWarnings = [logicWarningsMulti, logicWarningsSingle].some(
     array => array.length > 0,
   );
 
-  if (isFetchingGetSubmissionLatest) return <LoadingContent />;
+  const isVerified =
+    isSubmissionAccepted(submission) || boxChecked || !hasWarnings;
 
-  console.log(
-    `${lei}-${year} file/submit info:`,
-    actualDataGetSubmissionLatest,
-  );
-  // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unnecessary-condition
+  const canContinue = !formSubmitLoading && !errorSubmissionFetch && isVerified;
 
-  console.log(
-    'errors warnings summary:',
-    getErrorsWarningsSummary(actualDataGetSubmissionLatest),
-  );
-
-  const onPreviousClick = (): void => {
-    navigate(`/filing/${year}/${lei}/errors`);
+  /* 
+    Event handlers 
+  */
+  const onClickCheckbox = (): void => {
+    setBoxChecked(!boxChecked);
   };
 
-  const onNextClick = (): void => {
-    navigate(`/filing/${year}/${lei}/contact`);
+  const onFormSubmit = async (): Promise<void> => {
+    const nextPage = `/filing/${year}/${lei}/contact`;
```

Comment:
Can we pull these API calls into separate file in the[ API requests folder](https://github.com/cfpb/sbl-frontend/blob/434-warnings-part2/src/api/requests/)? I really liked that folder when we used it consistently. It'll be easier for future testing.

Reply:
Whoops! Yep this is fine as is, and we can talk centralizing the logic in another PR @meissadia.